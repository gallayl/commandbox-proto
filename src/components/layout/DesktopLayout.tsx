import CssBaseline from '@material-ui/core/CssBaseline'
import React from 'react'
import snLogo from '../../assets/sensenet_logo_transparent.png'
import { DesktopAppBar } from '../appbar/DesktopAppBar'
import { DesktopDrawer } from '../drawer/DesktopDrawer'

export const DesktopLayout: React.FunctionComponent = props => {
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        position: 'fixed',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
      }}>
      <CssBaseline />
      <DesktopAppBar />
      <div
        style={{
          flexGrow: 1,
          display: 'flex',
          flexDirection: 'row',
          height: '100%',
        }}>
        <DesktopDrawer />
        <div
          style={{
            display: 'flex',
            flexGrow: 1,
            backgroundPosition: 'center',
            backgroundRepeat: 'no-repeat',
            backgroundImage: `url(${snLogo})`,
          }}>
          {props.children}
        </div>
      </div>
    </div>
  )
}
