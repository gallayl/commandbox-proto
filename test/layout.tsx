import { shallow } from 'enzyme'
import React from 'react'
import { DesktopLayout } from '../src/components/Layout/DesktopLayout'

describe('Layout', () => {
  it('Matches snapshot', () => {
    const l = shallow(<DesktopLayout />)
    expect(l).toMatchSnapshot()
  })
})
