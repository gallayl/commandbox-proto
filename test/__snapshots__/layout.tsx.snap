// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Matches snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "left": 0,
      "position": "fixed",
      "top": 0,
      "width": "100%",
    }
  }
>
  <WithStyles(CssBaseline) />
  <Connect(DesktopAppBar) />
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexGrow": 1,
      }
    }
  >
    <withRouter(Connect(DesktopDrawer)) />
    <div
      style={
        Object {
          "display": "flex",
          "flexGrow": 1,
        }
      }
    />
  </div>
</div>
`;
